00_Base/src/model/ocpi.response.ts

```diff
@@ -19,34 +19,40 @@ export enum OcpiResponseStatusCode {
   HubConnectionProblem = 4003,
 }
 
+export enum OcpiResponseProps {
+  status_code = 'status_code',
+  status_message = 'status_message',
+  timestamp = 'timestamp',
+  data = 'data',
+}
+
 export class OcpiResponse<T> {
   @Enum(OcpiResponseStatusCode, 'OcpiResponseStatusCode')
   @IsNotEmpty()
-  status_code!: OcpiResponseStatusCode;
+  [OcpiResponseProps.status_code]!: OcpiResponseStatusCode;
   /**
    *
    * @type {string}
    * @memberof OcpiResponseDTO
    */
   @IsString()
   @Optional()
-  status_message?: string;
+  [OcpiResponseProps.status_message]?: string;
 
   /**
    *
    * @type {string}
    * @memberof OcpiResponseDTO
    */
   @IsDate()
-  timestamp!: Date;
+  [OcpiResponseProps.timestamp]!: Date;
 
   /**
    *
    * @type {object}
    * @memberof OcpiResponseDTO
    */
-  @Optional()
```

Comment:
Why is optional removed?

Reply:
Good question - so it turns out with Reflect and inheritance, what I saw that if lets say we have our structure:
```
export class OcpiResponse<T> {
  @Optional()
  data?: T;
}
```
and then we have:
```
export class CdrResponse extends OcpiResponse<CdrDTO> {
  Optional(false)
  data!: CdrDTO[]
}
```

then unfortunately it does not behave as you would expect. The optional in the derived class doesnt override things as you would assume it would. I have faced this in a few places and found that it is best to annotate the derived classes and keep base classes in simple TS without annotations unless there is an annotation that will truly be shared and enabled by all derived classes....

Removing this Optional because it is simply unnecessary because our derived classes already do define whether data is optional or not, so our base class is acting more like an interface here. Does that make sense? I can try and clarify further